app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -344,6 +344,16 @@ def uuid_from_href(ems_ref)
       URI(ems_ref).path.split('/').last
     end
 
+    def find_mac_address_on_network(nics, network, log)
+      ext_management_system.with_provider_connection(VERSION_HASH) do |connection|
+        nic = nics.detect do |n|
+          connection.follow_link(n.vnic_profile).network.id == network[:id]
```

Comment:
It isn't currently possible. In the future it will be possible, using the server side link following:

```ruby
nics_service = ...
nics_with_profiles = nics_service.list(follow='vnic_profile')
```